File: src/backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/dms/revision/dto/RevisionHistoryDto.java
package pt.ulisboa.tecnico.rnl.dei.dms.revision.dto;

import java.time.LocalDateTime;

import pt.ulisboa.tecnico.rnl.dei.dms.person.dto.PersonDto;
import pt.ulisboa.tecnico.rnl.dei.dms.evaluation.dto.TestDto;
import pt.ulisboa.tecnico.rnl.dei.dms.revision.domain.RevisionHistory;

public record RevisionHistoryDto(long id, PersonDto student, TestDto test, String status, LocalDateTime dateTime) {
    public RevisionHistoryDto(RevisionHistory history) {
        this(
            history.getId(),
            new PersonDto(history.getRevision().getStudent()),
            new TestDto(history.getRevision().getTest()),
            history.getStatus().toString(),
            history.getDateTime()
        );
    }
}

File: src/frontend/src/models/RevisionHistoryDto.ts
import type TestDto from "./TestDto";
import type PersonDto from "./PersonDto";

export default class RevisionHistoryDto {
  id?: number
  student?: PersonDto
  test?: TestDto
  status?: string
  dateTime?: string

  constructor(obj?: Partial<RevisionHistoryDto>) {
    Object.assign(this, obj)
  }
}